include/libuvc/libuvc_internal.h

```diff
@@ -291,6 +291,7 @@ struct uvc_device_handle {
   /** Whether the camera is an iSight that sends one header per frame */
   uint8_t is_isight;
   uint32_t claimed;
+  uint8_t detached;
```

Comment:
better introduce additional flag: `should_detach`
and don't change the prototype of `uvc_claim_if`